app/src/main/java/org/asteroidos/sync/ble/MediaService.java

```diff
@@ -138,6 +183,29 @@ public void onEvent(ReadWriteEvent e) {
                                 mMediaController.getTransportControls().pause();
                             }
                              break;
+                        case MEDIA_COMMAND_VOLUME:
+                            if (mMediaController.getPlaybackInfo() != null) {
+                                if (data[1] != mVolume) {
+                                    int delta = Math.abs(mVolume - data[1]);
+                                    int deviceDelta = 100/mMediaController.getPlaybackInfo().getMaxVolume();
+                                    // Change in volume is smaller than the device volume step (i.e. volume won't change)
+                                    // Always change volume, in this case just use the Android
+                                    if (delta < deviceDelta) {
```

Comment:
this is obviously a hacky workaround but given your explanation this sounds reasonable.
I would make this workaround more clear in the comment for future code edits e.g.:
```
// if the change in volume is smaller than the device volume step (i.e. volume won't change)
// increase or decrease one step anyway to improve UX
```